mlte/spec/spec.py

```diff
@@ -42,15 +43,15 @@ class Spec(Artifact):
     and the results of measurement evaluation and validation.
     """
 
-    def __init__(
-        self, identifier: str, properties: Dict[Property, Dict[str, Condition]]
-    ):
+    def __init__(self, properties: Dict[Property, Dict[str, Condition]]):
         """
         Initialize a Spec instance.
 
         :param properties: The collection of properties that compose the spec, with their conditions keyed by measurement id.
         :type properties: List[Property]
         """
+        # Standard identifier. Good enough?
+        identifier = DEFAULT_SPEC_ID
```

Comment:
Thank you for pointing out this update in PR comments @sebastian-echeverria. Here are my thoughts on this:
- I think in the typical use case, there will only ever be one spec for each (namespace, model, version). The same is true for some other artifacts: negotiation card, validated spec, and report
- However, I would like to be able to support multiple artifacts of each type in the event that they are needed
- I'd like to handle this by doing something similar to what you have here; the four types of artifacts that I mentioned above should accept an `identifier` argument on construction, like any other artifact that has multiple instances (e.g. `Value`)
- However, we also provide a default argument, similar to what you have here, that is `default` followed by the artifact type
- For instance, for `Spec`, it would be `default.spec`

Let me know what you think of this proposal

Reply:
I agree with the default idea for most artifacts, but allowing for different ones if needed. I agree about the default.artifactype for the default value, sure. Identifier is the first param in the artifact constructors, so if it has a default value as an argument, all other arguments must have a default value as well. I can add this for Spec and ValidatedSpec.